app/connectors/schema_mapping.py

```
@@ -0,0 +1,59 @@
+    default_schema_version: int = 1
+
+    def resolve_event_type(
+        self, row: dict[str, Any], operation: str
+    ) -> str | None:
+        """Evaluate rules in priority order; return first match."""
+        context = {"row": row, "operation": operation, **row}
+        for rule in sorted(self.event_type_rules, key=lambda r: -r.priority):
+            if rule.condition is None:
+                return rule.event_type
+            try:
+                if eval(rule.condition, {"__builtins__": {}}, context):
```

Ruff (0.15.5)

[error] Use of possibly insecure function; consider using `ast.literal_eval`

(S307)